package/MDAnalysis/core/universe.py

```diff
@@ -627,26 +636,42 @@ def transfer_to_memory(self, start=None, stop=None, step=None,
         from ..coordinates.memory import MemoryReader
 
         if not isinstance(self.trajectory, MemoryReader):
-            # Try to extract coordinates using Timeseries object
-            # This is significantly faster, but only implemented for certain
-            # trajectory file formats
-            try:
-                coordinates = self.trajectory.timeseries(
```

Comment:
Is it true that we do not have dedicated C-level timeseries anymore and thus the `self.trajectory.timeseries` was essentially just doing what happened in the `except` block? Or are there still cases where the `timeseries` approach would be beneficial?

Reply:
Only DCDReader ever had a fast timeseries (and still does).  It didn't return dimensions information though, which is needed.  We could change `DCD.timeseries` to have `DCD.timeseries(...., dimensions=True)` to then return both coordinates and dimensions.

Then maybe the `base.Reader` could implement the entire `else:` branch below (I started this in #1400 once...)

So yeah maybe I should try and use the fast DCD routine here still since it still exists, then manually grab the dimensions..